service/matching/taskListManager.go

```diff
@@ -338,6 +343,7 @@ func (c *taskListManagerImpl) Start() error {
 	c.taskWriter.Start()
 	c.signalNewTask()
 	go c.getTasksPump()
+	go c.checkIdle()
```

Comment:
Why do we need a separate goroutine for idle check?  why can't be done in the same pump like we do for zero pollers check?

Reply:
Changed, previously because I try to avoid this check potentially block getTaskPump